### PR TITLE
Fix conformance test script to allow building from k8s release

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Fix conformance test script to allow building from k8s release. Some bash improvements, fixed indents. Also adds documentation for running locally.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nader-ziada** [comment]
is there any doc that lists the skip flags? can't find that

**Comment by CecileRobertMichon** [comment]
@nader-ziada there isn't, I can add one

**Comment by nader-ziada** [comment]
I think it would be useful even as comments in the script so users can figure out how to run it. Thanks

File: docs/development.md [hunk 1]
```diff
@@ -289,7 +289,30 @@ To run E2E locally, set `AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET`, `AZURE_SUBSCRI
 ./scripts/ci-e2e.sh
 ```
 
-You can optionally set `AZURE_SSH_PUBLIC_KEY_FILE` to use your own ssh key.
+You can optionally set `AZURE_SSH_PUBLIC_KEY_FILE` to use your own SSH key.
+
+#### Conformance Testing
```
Comment (CecileRobertMichon) [inline docs/development.md:294]: @nader-ziada pls take a look. Thanks for the suggestion!

**Comment by nader-ziada** [comment]
Thanks @CecileRobertMichon 

/lgtm

**Pull request merged**